Improve handling of analogue input events

ReFerox opened the pull request:
### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

Replacing all calls to emplace with try_emplace

davidebeatrici commented:
```
D:/a/1/s/src/mumble/GlobalShortcut_win.cpp(442): error C2220: the following warning is treated as an error
D:/a/1/s/src/mumble/GlobalShortcut_win.cpp(442): warning C4834: discarding return value of function with [[nodiscard]] attribute
D:/a/1/s/src/mumble/GlobalShortcut_win.cpp(458): warning C4834: discarding return value of function with [[nodiscard]] attribute
D:/a/1/s/src/mumble/GlobalShortcut_win.cpp(464): warning C4834: discarding return value of function with [[nodiscard]] attribute
```

davidebeatrici commented:
@Krzmbrzl @Hartmnt Do we perhaps want to use [`std::ignore`](https://en.cppreference.com/w/cpp/utility/tuple/ignore) instead?

Krzmbrzl commented:
I vote for keeping the logging and extending this PR to filter out all input events from analog input devices right where we receive them (that is, before attempting to queue them).

This should prevent log spamming as we shouldn't run into full queues anymore but if we do, we at least get notified.

This has the additional benefit that regular events won't be dismissed just because the queue happens to be full of meaningless events from analog input devices.

davidebeatrici commented:
We can pick it up further, but would like to give you the chance since you're doing great!

This is the function responsible for pushing the raw input messages into the queue: https://github.com/mumble-voip/mumble/blob/41e381db6241cd6bfb02061e7096f71266c7937d/src/mumble/GlobalShortcut_win.cpp#L427-L472

In the `RIM_TYPEHID` we should filter out all analog input events, at least for the time being. Once we figure out where the bottleneck is exactly, we can then also integrate the patch you provided in #6696 to add support for analog input (in a future pull request).

ReFerox commented:
Thanks again for the feedback. I removed the [[maybe_unused]] and i added a filter for the analog signals. It should check if the signal of the analog device is a 8 or 16 bit signed or unsinged value and check values according to this.

davidebeatrici commented:
The function looks really interesting... how does it distinguish between buttons and analog things? Is the HID report format documented somewhere?

ReFerox commented:
Hi @davidebeatrici 

What is the current status on this PR? Is it now further up to you to intergrate this further in mumble? Or is there something else must do? 

Please let me know, thanks :)

davidebeatrici commented:
Sorry for the delay, I'll test your code as soon as possible!

ReFerox commented on src/mumble/GlobalShortcut_win.cpp:
```diff
@@ -424,6 +424,57 @@ void GlobalShortcutWin::run() {
+	}
+
+	// Check unsigned 16-bit values
+	for (size_t i = 0; i + 1 < data.size(); i += 2) {
+		uint16_t u16 = static_cast<uint16_t>(data[i]) | (static_cast<uint16_t>(data[i + 1]) << 8);
+		debug << " u16:" << u16;
+		if (u16 > 512 && u16 < 64512) {
+			++midrangeCount;
+		}
+	}
+
+	// Check signed 16-bit values
+	for (size_t i = 0; i + 1 < data.size(); i += 2) {
+		int16_t s16 = static_cast<int16_t>(static_cast<uint16_t>(data[i]) | (static_cast<uint16_t>(data[i + 1]) << 8));
+		debug << " s16:" << s16;
+		if (s16 > -16384 && s16 < 16384) {
+			++midrangeCount;
+		}
```
Merched

ReFerox commented on src/mumble/GlobalShortcut_win.cpp:
```diff
@@ -424,6 +424,57 @@ void GlobalShortcutWin::run() {
+		}
+	}
+
+	qDebug("GlobalShortcutWin: HID report analysis: %s", debug.str().c_str());
```
Removed
